src/plugins/intel_npu/src/plugin/npuw/llm_infer_request.cpp

```diff
@@ -0,0 +1,239 @@
+            //         m_kvcache_desc.max_prompt_size - m_kvcache_desc.num_stored_tokens,
+            //         0, m_kvcache_desc.num_stored_tokens);
+        }
+        LOG_DEBUG("LLMInferRequest::infer_generate(): Prepare attention mask pattern.");
+        auto* attention_mask_data = m_kvcache_request->get_tensor(
+            m_kvcache_in_ports.at("attention_mask"))->data<int64_t>();
+        attention_mask_data[m_kvcache_desc.total_size - 1] = 1;
+
+        m_need_copy_kvcache = false;
+    }
+
+    auto kv_input_ids = m_kvcache_request->get_tensor(m_kvcache_in_ports.at("input_ids"));
+    std::copy_n(input_ids->data<int64_t>(), input_ids->get_size(), kv_input_ids->data<int64_t>());
+
+    auto kv_attn_mask = m_kvcache_request->get_tensor(m_kvcache_in_ports.at("attention_mask"));
+    std::copy_n(attention_mask->data<int64_t>(), attention_mask->get_size(), kv_attn_mask->data<int64_t>());
+
+    auto kv_pos_ids = m_kvcache_request->get_tensor(m_kvcache_in_ports.at("position_ids"));
+    std::copy_n(position_ids->data<int64_t>(), position_ids->get_size(), kv_pos_ids->data<int64_t>());
+
+    m_kvcache_request->infer();
+    m_logits = m_kvcache_request->get_tensor(m_kvcache_out_ports.at("logits"));
+    m_kvcache_desc.num_stored_tokens += 1;
+
+    LOG_DEBUG("Write KV-cache for the new token to the correct input position for next iteration.");
+    const auto kStartOutputKVCacheLayers = 1u;
+    const auto& kvcache_compiled = m_kvcache_request->get_compiled_model();
+    for (int i = 0; i < kvcache_compiled->outputs().size() - 1; ++i) {
+        const auto& output_name = kvcache_compiled->outputs()[kStartOutputKVCacheLayers + i].get_any_name();
+        const auto& input_name = std::regex_replace(output_name, std::regex("present"), "past_key_values");
+        auto kvcache_in_tensor = ov::make_tensor(m_kvcache_request->get_tensor(m_kvcache_in_ports.at(input_name)));
+        auto kvcache_in_slice = make_tensor_slice(
+            kvcache_in_tensor, m_kvcache_desc.dim, m_kvcache_desc.num_stored_tokens - 1, m_kvcache_desc.num_stored_tokens
+        );
+        auto kvcache_out_tensor = ov::make_tensor(m_kvcache_request->get_tensor(m_kvcache_out_ports.at(output_name)));
+        kvcache_out_tensor.copy_to(kvcache_in_slice);
+    }
+    LOG_DEBUG("Done");
+}
+
+void ov::npuw::LLMInferRequest::infer() {
+    const auto& inputs = get_inputs();
+
+    auto input_ids      = get_tensor(inputs[0]);
+    auto attention_mask = get_tensor(inputs[1]);
+    auto position_ids   = get_tensor(inputs[2]);
+
+    if (input_ids->get_size() != 1) {
+        infer_prefill(input_ids, attention_mask, position_ids);
+    } else {
+        infer_generate(input_ids, attention_mask, position_ids);
+    }
+}
+
+ov::SoPtr<ov::ITensor> ov::npuw::LLMInferRequest::get_tensor(const ov::Output<const ov::Node>& port) const {
+    // NB: If asked for logits...
+    if (port == get_outputs()[0]) {
+        return m_logits;
+    }
```

Comment:
I assume because this vector is taken from prefill model firstly and from kvcache one other times, so for our `LLMInferRequest` we should return different logits, but might be we can just set them as tensor in ourselves